### PR TITLE
✨ Add support to re-scaffolds projects with  alpha generate subcommand to deployImage plugin

### PR DESCRIPTION
**Pull request by yyy1000**
Description:
Support Deploy Plugin migration in the alpha generate subcommand.

### CODE REVIEW

File: pkg/rescaffold/migrate.go [hunk 1]
```diff
@@ -172,6 +176,41 @@ func migrateGrafanaPlugin(store store.Store, src, des string) error {
...
+		if errors.As(err, &config.PluginKeyNotFoundError{}) {
+			log.Printf("deploy-image plugin is not found, skip the migration")
+			return nil
+		}
+		return fmt.Errorf("failed to decode deploy-image plugin config %v", err)
+	}
+	// Here it is always empty, I don't know why
+	// if len(deployImagePlugin.Resources) == 0 {
+	// 	log.Printf("empty deploy-image plugin config, skip the migration")
+	// }
```
Comment (yyy1000) [inline pkg/rescaffold/migrate.go:194]: When DecodePluginConfig, the deployImagePlugin is always empty. I don't know what happened with it. :(

File: test/e2e/alphagenerate/generate_test.go [hunk 2]
```diff
@@ -197,7 +210,24 @@ func ReGenerateProject(kbc *utils.TestContext) {
...
 	ExpectWithOffset(1, fileContainsExpr).To(BeTrue())
 
-	By("checking if the project file was generated with the expected controller")
+	By("checking if the project file was generated with the expected deploy-image plugin fields")
+	var deployImagePlugin = "deploy-image.go.kubebuilder.io/v1-alpha"
+	fileContainsExpr, err = pluginutil.HasFileContentWith(
+		filepath.Join(kbc.Dir, "testdir2", "PROJECT"), deployImagePlugin)
+	Expect(err).NotTo(HaveOccurred())
+	Expect(fileContainsExpr).To(BeTrue())
+	var deployImagePluginFields = `      kind: Memcached
```
Comment (Kavinjsir) [inline test/e2e/alphagenerate/generate_test.go:219]:
```suggestion
	var deployImagePluginFields = `kind: Memcached
```
You can probably drop the spaces here?

Comment (yyy1000) [reply test/e2e/alphagenerate/generate_test.go:219]: Good idea.

**Review by Kavinjsir — APPROVED** [review]
LGTM! 👍🏼

**Review by camilamacedo86 — APPROVED** [review]

**Comment by camilamacedo86** [comment]
/hold 

@yyy1000 we must squash the commits. COuld you please do that?

**Review by Kavinjsir — APPROVED** [review]

**Pull request merged**